### PR TITLE
GIT Mass operations

### PR DESCRIPTION
**Pull request by slemrmartin**
Scripts iterate over core and plugins and performs useful mass operations:
* Mass checkout master & pull
* Mass rebase (checkout master, pull, checkout back and rebase)
* Mass list branches
* Mass list changes
* Mass cleanup (delete branches merged to master)

### CODE REVIEW

**Comment by slemrmartin** [comment]
@martinpovolny pls review

File: git_utils/config.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# You can override these values in file config.dev.sh
+homedir=~
+eval homedir=${homedir}
```
Comment (himdel) [inline git_utils/config.sh]: that's just `$HOME`

File: git_utils/cleanup.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    then
+        remote_merged=$(git branch -r --merged master | sed 's/ *origin\///' | grep -v 'master$')
+        if [[ -z ${remote_merged} ]]; then
+           echo "No remote branches to delete"
+        else
+            # Remove remote fully merged branches
+            echo ${remote_merged} | xargs -I% git push origin :%
+        fi
+    fi
+	cd ..
```
Comment (himdel) [inline git_utils/cleanup.sh]: bad indent

**Comment by himdel** [comment]
@slemrmartin The bash files are a mix of tabs and spaces for indentation. Please fix it to use something consistent ;)

File: git_utils/config.sh (hunk 1 above)
Comment (slemrmartin) [reply git_utils/config.sh]: on mac too? ok

File: git_utils/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+List of repositories are either generated automatically (`$custom_repo_list == 0`) or you can define your own array manually
+(`$custom_repo_list == 1`). See config for example. 
+
+### Scripts
+
+- [pull.sh](pull.sh): Checkouts all unchanged repos to master and pulls changes
+- [rebase.sh](rebase.sh): For all unchanged repos does the same as pull.sh and then rebases your current branch
+- [list-branches.sh](list-branches.sh): Prints current branches in all repos
+- [list-changes.sh](list-changes.sh): Prints changes in all repos 
+- [cleanup.sh](scripts/git/cleanup.sh): Deletes branches merged to master
```
Comment (gmcculloug) [inline git_utils/README.md]: The link to `cleanup.sh` is broken.  You do not need `scripts/git/`.

**Comment by himdel** [comment]
(Not opposing this as is, but I'd love to see this structured so that you have N scripts which just work on the current repo, and 1 script that will allow you to run any of those N scripts in all the configured repos. (Instead of doing the looping in each.))

**Comment by slemrmartin** [comment]
@himdel done

**Pull request merged**